packages/components/src/internal/components/assay/AssayImportSubMenuItem.tsx

```diff
@@ -29,15 +34,16 @@ export class AssayImportSubMenuItemImpl extends PureComponent<Props & InjectedAs
     };
 
     getItems = (): ISubItem[] => {
-        const { assayModel, model, providerType } = this.props;
+        const { assayModel, model, providerType, queryModel } = this.props;
+        let importItems;
 
-        return getImportItemsForAssayDefinitions(assayModel, model, providerType).reduce((subItems, href, assay) => {
-            subItems.push({
-                text: assay.name,
-                href,
-            });
-            return subItems;
-        }, []);
+        if (queryModel !== undefined) {
+            importItems = getImportItemsForAssayDefinitionsQM(assayModel, queryModel, providerType);
+        } else {
+            importItems = getImportItemsForAssayDefinitions(assayModel, model, providerType);
+        }
+
+        return importItems.map((subItems, href, assay) => ({ text: assay.name, href }));
```

Comment:
`subItems` was part of the `reduce` but looks like it isn't used here in the `map`

Reply:
Yeah, this is not correct, I think I messed something up and will figure this out. The reduce args don't make sense either, but this is running map on an Immutable OrderedMap so maybe the args are different than I would expect for any other iterable.

Reply:
Ok, I reverted it back to the reduce statement after reading up on OrderedMaps.